src/LuaPiGui.cpp

```diff
@@ -0,0 +1,1105 @@
+// 	else if(!name.compare("radial_in"))
+// 		marker = wv->GetRadialOutIndicator();
+// 	else if(!name.compare("radial_out"))
+// 		marker = wv->GetRadialInIndicator();
+// 	else if(!name.compare("away_from_frame"))
+// 		marker = wv->GetAwayFromFrameIndicator();
+// 	else if(!name.compare("combat_target"))
+// 		marker = wv->GetCombatTargetIndicator();
+// 	else if(!name.compare("combat_target_lead"))
+// 		marker = wv->GetCombatTargetLeadIndicator();
+// 	else if(!name.compare("maneuver"))
+// 		marker = wv->GetManeuverIndicator();
+
+// 	else
+// 		// TODO: error
+// 		return 0;
+// 	vector2f position = marker->pos - vector2f(Graphics::GetScreenWidth()/2,Graphics::GetScreenHeight()/2);
+// 	vector2f direction = position.Normalized();
+// 	std::string side = marker->side == INDICATOR_HIDDEN ? "hidden" : (marker->side == INDICATOR_ONSCREEN ? "onscreen" : "other");
+// 	lua_pushstring(l, side.c_str());
+// 	LuaTable dir(l);
+// 	dir.Set("x", direction.x);
+// 	dir.Set("y", direction.y);
+// 	dir.Set("z", 0);
+// 	LuaTable pos(l);
+// 	pos.Set("x", marker->pos.x);
+// 	pos.Set("y", marker->pos.y);
+// 	pos.Set("z", 0);
+// 	//	  Output("forward: %f/%f %s\n", marker->pos.x, marker->pos.y, side.c_str());
+// 	return 3;
+// }
+
+static int l_pigui_add_text(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 center = luaL_checkImVec2(l, 1);
+	ImColor color = luaL_checkImColor(l, 2);
+	std::string text = luaL_checkstring(l, 3);
+	draw_list->AddText(center, color, text.c_str());
+	return 0;
+}
+
+static int l_pigui_add_triangle(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 a = luaL_checkImVec2(l, 1);
+	ImVec2 b = luaL_checkImVec2(l, 2);
+	ImVec2 c = luaL_checkImVec2(l, 3);
+	ImColor col = luaL_checkImColor(l, 4);
+	float thickness = luaL_checknumber(l, 5);
+	draw_list->AddTriangle(a, b, c, col, thickness);
+	return 0;
+}
+
+static int l_pigui_add_rect(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 a = luaL_checkImVec2(l, 1);
+	ImVec2 b = luaL_checkImVec2(l, 2);
+	ImColor col = luaL_checkImColor(l, 3);
+	float rounding = luaL_checknumber(l, 4);
+	int round_corners = luaL_checkinteger(l, 5);
+	float thickness = luaL_checknumber(l, 6);
+	draw_list->AddRect(a, b, col, rounding, round_corners, thickness);
+	return 0;
+}
+
+static int l_pigui_add_bezier_curve(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 a = luaL_checkImVec2(l, 1);
+	ImVec2 c0 = luaL_checkImVec2(l, 2);
+	ImVec2 c1 = luaL_checkImVec2(l, 3);
+	ImVec2 b = luaL_checkImVec2(l, 4);
+	ImColor col = luaL_checkImColor(l, 5);
+	float thickness = luaL_checknumber(l, 6);
+	int num_segments = luaL_checkinteger(l, 7);
+	draw_list->AddBezierCurve(a, c0, c1, b, col, thickness, num_segments);
+	return 0;
+}
+
+static int l_pigui_add_image(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImTextureID id = reinterpret_cast<ImTextureID>(luaL_checkinteger(l, 1));
```

Comment:
I'd need to check the matching lua code, but shouldn't this be userdata ? Unless you want the lua side to be able to do number-like stuff with it.